compiler/qsc_qasm3/src/semantic/lowerer.rs

```diff
@@ -420,7 +437,12 @@ impl Lowerer {
         }
 
         let lhs = self.lower_indexed_ident_expr(lhs);
-        let rhs = self.lower_expr(rhs);
+        let rhs = match rhs {
+            syntax::ValueExpr::Expr(expr) => {
+                self.lower_expr_with_target_type(Some(expr), ty, stmt.span)
+            }
+            syntax::ValueExpr::Measurement(measure_expr) => self.lower_measure_expr(measure_expr),
```

Comment:
I think we need an additional call to cast to tartget type in the measure case as the lhs might not be a bit and we need to implicitly cast.